devlib/target.py

```diff
@@ -517,21 +517,26 @@ def path_kind(path):
             src_excep = HostError
             src_path_kind = host_paths_kind
 
-            dst_mkdir = once(self.makedirs)
+            _dst_mkdir = once(self.makedirs)
             dst_path_join = self.path.join
             dst_paths_kind = target_paths_kind
             dst_remove_file = once(functools.partial(self.remove, as_root=as_root))
         elif action == 'pull':
             src_excep = TargetStableError
             src_path_kind = target_paths_kind
 
-            dst_mkdir = once(functools.partial(os.makedirs, exist_ok=True))
+            _dst_mkdir = once(functools.partial(os.makedirs, exist_ok=True))
             dst_path_join = os.path.join
             dst_paths_kind = host_paths_kind
             dst_remove_file = once(os.remove)
         else:
             raise ValueError('Unknown action "{}"'.format(action))
 
+        def dst_mkdir(path):
+            # path is None if dirname('foo.bar')
```

Comment:
Could this comment be clarified? I'm not sure what it is meant to mean.

Reply:
@marcbonnici  Does it look better?

Reply:
Great thanks, could you fix the typo in the commit message `/s/beahviour/behaviour/` and then it should be good to go.

Reply:
Apologies, I missed your previous comment. This is done now.